include/lsst/afw/table/FieldBase.h

```diff
@@ -211,7 +211,7 @@ struct FieldBase< Array<U> > {
 
     template <typename Derived>
     void setValueDeep(Element * p, ndarray::ExpressionBase<Derived> const & value) const {
-        if (value.template getSize<0>() != _size) {
+        if (value.template getSize<0>() != static_cast<std::size_t>(_size)) {
```

Comment:
The signature of this `FieldBase` instantiation's constructor needs to match that of another instantiation for which negative values at schema-definition time are used to indicate a variable-size array field (i.e. a different size for each record, so this needs to stay as a signed `int`. Switching to 64-bits might be worthwhile, but I'd rather not do that on this ticket.